### ISSUE TITLE
Wall profile interferes with socket operations

### ISSUE THREAD

**Issue by silvpol** [GoogleCloudPlatform/cloud-profiler-java#23]
Hi

I have a Java application running on GKE and whenever wall profile is triggered a series of warnings regarding socket connectivity is logged.
Disabling profiler altogether or setting thread cut-off to a low value makes the warnings disappear at the cost of disabling wall profiling. This is not likely due to thread count, it's in mid 40s.

Versions:
Stackdriver Profiler Java agent: cloud-profiler-java-agent_20200504_RC00
Java: 11.0.6 OpenJDK from Docker openjdk:11-jre container

Example log entries:

````
Info 2020-05-11 16:41:49.690 BST "Initialized deployment: project_id=uk-non-prod-6ae1c9dc, service=screen, service_version=35.1.0-rc11, zone_name=europe-west2-b"
Info 2020-05-11 16:41:49.690 BST "Creating a new profile via profiler service"
Info 2020-05-11 16:41:52.852 BST "Profile created: cpu projects/uk-non-prod-6ae1c9dc/profiles/4e358c73ef33371e"
Info 2020-05-11 16:42:02.955 BST "Collected a profile: total count=5, weight=50000000"
Info 2020-05-11 16:42:02.955 BST "Uploading 4920 bytes of 'cpu' profile data"
Info 2020-05-11 16:42:03.204 BST "Initialized deployment: project_id=uk-non-prod-6ae1c9dc, service=screen, service_version=35.1.0-rc11, zone_name=europe-west2-b"
Info 2020-05-11 16:42:03.204 BST "Creating a new profile via profiler service"
Info 2020-05-11 16:42:03.325 BST "Profile created: heap projects/uk-non-prod-6ae1c9dc/profiles/64ff9552bd0cf759"
Info 2020-05-11 16:42:03.338 BST "Uploading 31184 bytes of 'heap' profile data"
Info 2020-05-11 16:42:03.644 BST "Initialized deployment: project_id=uk-non-prod-6ae1c9dc, service=screen, service_version=35.1.0-rc11, zone_name=europe-west2-b"
Info 2020-05-11 16:42:03.645 BST "Creating a new profile via profiler service"
Info 2020-05-11 16:42:23.635 BST "Profile created: cpu projects/uk-non-prod-6ae1c9dc/profiles/756ddea54ac007eb"
Info 2020-05-11 16:42:33.738 BST "Collected a profile: total count=7, weight=70000000"
Info 2020-05-11 16:42:33.739 BST "Uploading 5015 bytes of 'cpu' profile data"
Info 2020-05-11 16:42:33.991 BST "Initialized deployment: project_id=uk-non-prod-6ae1c9dc, service=screen, service_version=35.1.0-rc11, zone_name=europe-west2-b"
Info 2020-05-11 16:42:33.992 BST "Creating a new profile via profiler service"
Info 2020-05-11 16:42:34.114 BST "Profile created: wall projects/uk-non-prod-6ae1c9dc/profiles/3109fbcf06d1e31f"
Warning 2020-05-11 16:42:34.116 BST "JGRP000006: failed accepting connection from peer null: java.net.SocketTimeoutException: Accept timed out"
...
Removed list of similar messages around 250ms apart
...
Warning 2020-05-11 16:42:43.603 BST "JGRP000006: failed accepting connection from peer null: java.net.SocketTimeoutException: Accept timed out"
Warning 2020-05-11 16:42:43.861 BST "JGRP000006: failed accepting connection from peer null: java.net.SocketTimeoutException: Accept timed out"
Info 2020-05-11 16:42:44.373 BST "Collected a profile: total count=1560, weight=399999999360"
Info 2020-05-11 16:42:44.374 BST "Uploading 3521 bytes of 'wall' profile data"
Info 2020-05-11 16:42:44.644 BST "Initialized deployment: project_id=uk-non-prod-6ae1c9dc, service=screen, service_version=35.1.0-rc11, zone_name=europe-west2-b"
Info 2020-05-11 16:42:44.644 BST "Creating a new profile via profiler service"
Info 2020-05-11 16:42:44.772 BST "Profile created: heap projects/uk-non-prod-6ae1c9dc/profiles/57f5d3651332d30b"
Info 2020-05-11 16:42:44.785 BST "Uploading 31283 bytes of 'heap' profile data"
Info 2020-05-11 16:42:45.029 BST "Initialized deployment: project_id=uk-non-prod-6ae1c9dc, service=screen, service_version=35.1.0-rc11, zone_name=europe-west2-b"
Info 2020-05-11 16:42:45.029 BST "Creating a new profile via profiler service"
Info 2020-05-11 16:43:17.851 BST "Profile created: cpu projects/uk-non-prod-6ae1c9dc/profiles/35b53af16308f8d7"
Info 2020-05-11 16:43:27.955 BST "Collected a profile: total count=6, weight=60000000"
Info 2020-05-11 16:43:27.956 BST "Uploading 7863 bytes of 'cpu' profile data"
Info 2020-05-11 16:43:28.202 BST "Initialized deployment: project_id=uk-non-prod-6ae1c9dc, service=screen, service_version=35.1.0-rc11, zone_name=europe-west2-b"
Info 2020-05-11 16:43:28.202 BST "Creating a new profile via profiler service"
Info 2020-05-11 16:43:52.464 BST "Profile created: heap projects/uk-non-prod-6ae1c9dc/profiles/60433945f634bf2a"
Info 2020-05-11 16:43:52.477 BST "Uploading 31283 bytes of 'heap' profile data"
````

**Comment by kalyanac** [comment]
Thanks for reporting this. Few questions to better understand the environment

How many services are enabled with Profiler? Do all of them exhibit the same behavior? 
Does this happen for every wall profile attempt?

**Comment by aalexand** [comment]
Might be related to https://bugs.openjdk.java.net/browse/JDK-8237858? Seems similar to https://github.com/jvm-profiling-tools/async-profiler/issues/295.

**Comment by silvpol** [comment]
@kalyanac 
> How many services are enabled with Profiler? Do all of them exhibit the same behavior?

We have 4, all enabled and all exhibit same problem.

> Does this happen for every wall profile attempt?

I have checked and looks like it, the messages are always printed between start and end of profiling messages.

@aalexand 
I looks very similar indeed, in an older version of our service I can see a stack trace:
````
JGRP000006: failed accepting connection from peer
java.net.SocketTimeoutException: Accept timed out
	at java.base/java.net.PlainSocketImpl.socketAccept(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.accept(Unknown Source)
	at java.base/java.net.ServerSocket.implAccept(Unknown Source)
	at java.base/java.net.ServerSocket.accept(Unknown Source)
	at org.jgroups.blocks.cs.TcpServer$Acceptor.run(TcpServer.java:110)
	at java.base/java.lang.Thread.run(Unknown Source)
````

**Comment by maxweber** [comment]
We use Google Cloud Profiler for our Clojure web application that runs on Google Kubernetes Engine. After enabling Cloud Profiler everything worked fine, except it was not longer possible to connect to a pod via [nrepl](https://nrepl.org/).

With the information found here, I guessed that the wall time profling might also interfere with nrepl's `java.net.Socket`. I used `-cprof_wall_num_threads_cutoff=0` to disable the wall time profiling, which solved our problem.
